lib/eslint.js

```diff
@@ -617,33 +657,45 @@ module.exports = (function() {
             Object.keys(config.rules).filter(function(key) {
                 return getRuleSeverity(config.rules[key]) > 0;
             }).forEach(function(key) {
-
-                var ruleCreator = rules.get(key),
-                    severity = getRuleSeverity(config.rules[key]),
-                    options = getRuleOptions(config.rules[key]),
-                    rule;
-
-                if (ruleCreator) {
-                    try {
-                        rule = ruleCreator(new RuleContext(
-                            key, api, severity, options,
-                            config.settings, config.ecmaFeatures
-                        ));
-
-                        // add all the node types as listeners
-                        Object.keys(rule).forEach(function(nodeType) {
-                            api.on(nodeType, timing.enabled
-                                ? timing.time(key, rule[nodeType])
-                                : rule[nodeType]
-                            );
-                        });
-                    } catch(ex) {
-                        ex.message = "Error while loading rule '" + key + "': " + ex.message;
-                        throw ex;
+                var ruleCreator,
+                    severity,
+                    options,
+                    rule,
+                    deprecationMsg;
+
+                if (!rules.get(key)) {
```

Comment:
FYI, this is what I think. Please make changes based on unanimous decision.
 
I see deprecation a way to tell the user that this rule would go away. But till the times its completely gone it will continue to work but also give out deprecation warning. That how concept of deprecation is supposed to work  and people understand it that way. Now if we are trying to do something different then its ok.